File: 6-kyu/numericals-of-a-string/swift/solution.swift
func numericals(_ str: String) -> String {
    var ret = String()
    var counts = [Character: Int]()
    for ch in str {
        if let currChCount = counts[ch] {
            counts[ch] = currChCount + 1
            ret += String(currChCount + 1)
        } else {
            ret += String(1)
            counts[ch] = 1
        }
    }
    return ret
}

File: 6-kyu/numericals-of-a-string/python/solution.py
def numericals(s):
    ret = ''
    counts = {}
    for i in range(len(s)):
        if s[i] not in counts:
            counts[s[i]] = 1
        else:
            counts[s[i]] += 1
        ret += str(counts[s[i]])
    return ret